cmd/ui/src/views/Explore/utils.ts

```diff
@@ -182,11 +184,12 @@ export const initGraphEdges = (graph: MultiDirectedGraph, edges: GraphEdges) =>
                 size: 3,
                 type: 'arrow',
                 label: edge.label,
-                color: '#000000C0',
+                color: defaultEdgeColor,
                 groupPosition: 0,
                 groupSize: 1,
                 exploreGraphId: edge.exploreGraphId || key,
                 forceLabel: true,
+                needsPerformance: edges.length > 250,
```

Comment:
This stands out to me - I'm curious how you came up with this heuristic?

Reply:
This was a line in the sand relative to the max nodes we allow to render. It would be useful to do some benchmarks for determining some measurable acceptable threshold for rendering performance on varied hardware. This number would also ideally discriminate between the amount of edges drawn using bezier curves vs linear edges since the former requires more computation but that involves looping through the edges again to update the attribute which I chose to avoid.
 
The positions of the arrowheads have been fixed so that there are no edges without arrowheads and so that there are not two arrowheads on one edge that should only have one in both the "fine" and "fast" methods so pivoting here will only affect arrowhead positioning on an edge when node to node distance is small and edge group offset is large.